core/src/main/java/org/apache/gravitino/storage/relational/service/CommonMetaService.java

```diff
@@ -36,25 +40,47 @@ public Long getParentEntityIdByNamespace(Namespace namespace) {
     Preconditions.checkArgument(
         !namespace.isEmpty() && namespace.levels().length <= 3,
         "Namespace should not be empty and length should be less than or equal to 3.");
+
     Long parentEntityId = null;
-    if (namespace.levels().length >= 1) {
+    if (namespace.levels().length == 1) {
```

Comment:
I can get the optimization logic here.
Can we further revise this logic into a `switch...case` for clarity?

Reply:
Okay